src/test/java/uk/ac/cam/cl/dtg/isaac/api/QuizFacadeIT.java

```diff
@@ -64,202 +120,3415 @@ public class QuizFacadeIT extends IsaacIntegrationTest {
+      assertEquals(Response.Status.OK.getStatusCode(), getQuizzesResponse.getStatus());
+
+      // check all quizzes are returned as available
+      @SuppressWarnings("unchecked") ResultsWrapper<QuizSummaryDTO> responseBody =
+          (ResultsWrapper<QuizSummaryDTO>) getQuizzesResponse.getEntity();
+      assertTrue(responseBody.getResults().stream().anyMatch(q -> q.getId().equals(QUIZ_TEST_QUIZ_ID)));
+      assertTrue(
+          responseBody.getResults().stream().anyMatch(q -> q.getId().equals(QUIZ_HIDDEN_FROM_ROLE_STUDENTS_QUIZ_ID)));
+      assertTrue(
+          responseBody.getResults().stream().anyMatch(q -> q.getId().equals(QUIZ_HIDDEN_FROM_ROLE_TUTORS_QUIZ_ID)));
+    }
+
+    /**
+     * Tests that quizzes with visibleToStudents=false and hiddenFromRoles=[TUTOR] are not considered available to a tutor.
+     */
+    @Test
+    public void getQuizzesAsTutor_returnsNonInvisibleToStudentOrHiddenFromRoleQuizzes()
+        throws NoCredentialsAvailableException, NoUserException, SegueDatabaseException,
+        AuthenticationProviderMappingException, IncorrectCredentialsProvidedException,
+        AdditionalAuthenticationRequiredException, InvalidKeySpecException, NoSuchAlgorithmException,
+        MFARequiredButNotConfiguredException {
+      // Arrange
+      // log in as Tutor, create request
+      LoginResult tutorLogin = loginAs(httpSession, TEST_TUTOR_EMAIL, TEST_TUTOR_PASSWORD);
+      HttpServletRequest getQuizzesRequest = createRequestWithCookies(new Cookie[] {tutorLogin.cookie});
+      replay(getQuizzesRequest);
+
+      // Act
+      // make request
+      Response getQuizzesResponse = quizFacade.getAvailableQuizzes(createNiceMock(Request.class), getQuizzesRequest);
+
+      // Assert
+      // check status code is OK
+      assertEquals(Response.Status.OK.getStatusCode(), getQuizzesResponse.getStatus());
+
+      // check invisible-to-student and hidden-from-tutor-role quizzes are not returned as available
+      @SuppressWarnings("unchecked") ResultsWrapper<QuizSummaryDTO> responseBody =
+          (ResultsWrapper<QuizSummaryDTO>) getQuizzesResponse.getEntity();
+      assertTrue(responseBody.getResults().stream().anyMatch(q -> q.getId().equals(QUIZ_TEST_QUIZ_ID)));
+      assertFalse(
+          responseBody.getResults().stream().anyMatch(q -> q.getId().equals(QUIZ_HIDDEN_FROM_ROLE_STUDENTS_QUIZ_ID)));
+      assertFalse(
+          responseBody.getResults().stream().anyMatch(q -> q.getId().equals(QUIZ_HIDDEN_FROM_ROLE_TUTORS_QUIZ_ID)));
+    }
+
+    @Test
+    public void getQuizzesAsStudent_returnsNonInvisibleToStudentQuizzes()
+        throws NoCredentialsAvailableException, NoUserException, SegueDatabaseException,
+        AuthenticationProviderMappingException, IncorrectCredentialsProvidedException,
+        AdditionalAuthenticationRequiredException, InvalidKeySpecException, NoSuchAlgorithmException,
+        MFARequiredButNotConfiguredException {
+      // Arrange
+      // log in as Tutor, create request
```

Comment:
It looks like this is logging in as a student, not a tutor.

Reply:
Iirc, that block of tests is the only one that existed before the ticket and wasn't replaced with a newer version. It looked to cover all the cases it needed to but I guess I'll need to give it another look over...